analyzers/tests/SonarAnalyzer.Test/Rules/ExceptionsShouldBeLoggedOrThrownTest.cs

```diff
@@ -34,6 +34,30 @@ public class ExceptionsShouldBeLoggedOrThrownTest
             .AddReferences(NuGetMetadataReference.MicrosoftExtensionsLoggingAbstractions())
             .Verify();
 
+    [TestMethod]
+    public void ExceptionsShouldBeLoggedOrThrown_Coalesce_CS() =>
+        builder
+            .AddSnippet("""
+                using System;
+                using Microsoft.Extensions.Logging;
+
+                public class Program
+                {
+                    public void Method(ILogger logger, object x)
+                    {
+                        try { }
+                        catch (SystemException e)
+                        {
+                            logger.LogError(e, "Message!");
+                            x = x ?? throw e;
```

Comment:
Curious: How is this not raising? 🤔 
I don't see where we exclude `CoalesceExpression`

Reply:
Yes, two nested problems. CoalesceExpression was not handled and the problem was not visible since the ThrowExpression was not handled :)